File: backend/app/schemas/technology_stack.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field


StackStatus = Literal["ready", "configured", "missing", "error"]


class TechnologyCheck(BaseModel):
    name: str
    category: str
    status: StackStatus
    details: str
    evidence: list[str] = Field(default_factory=list)


class TechnologyStackSummary(BaseModel):
    total: int
    ready: int
    configured: int
    missing: int
    errors: int
    production_ready_score: float = Field(ge=0, le=100)


class TechnologyStackResponse(BaseModel):
    generated_at: datetime
    environment: str
    summary: TechnologyStackSummary
    checks: list[TechnologyCheck]
    recommendations: list[str]

File: frontend/src/types/technology-stack.ts
export type TechnologyStatus = "ready" | "configured" | "missing" | "error";

export interface TechnologyCheck {
  name: string;
  category: string;
  status: TechnologyStatus;
  details: string;
  evidence: string[];
  remediation: string | null;
}

export interface TechnologyStackSummary {
  total: number;
  ready: number;
  configured: number;
  missing: number;
  errors: number;
  productionReadyScore: number;
}

export interface TechnologyStackResponse {
  summary: TechnologyStackSummary;
  checks: TechnologyCheck[];
  recommendations: string[];
  verifiedAt: string;
}
